### PR TITLE
Tidy-up logging

### PR DESCRIPTION
**Pull request by mtmk**
Added logging event ids and a few typo fixes.
Also removed `MinimumConsoleLoggerFactory` (saved a copy in an example) from core.

### CODE REVIEW

File: src/NATS.Client.Core/Internal/NatsReadProtocolProcessor.cs [hunk 1]
```diff
@@ -409,7 +409,7 @@ private async ValueTask<ReadOnlySequence<byte>> DispatchCommandAsync(int code, R
         else
         {
             // reaches invalid line, log warn and try to get newline and go to nextloop.
-            _logger.LogWarning("reaches invalid line.");
+            _logger.LogWarning(NatsLogEvents.Protocol, "Reaches invalid line");
```
Comment (Jarema) [inline src/NATS.Client.Core/Internal/NatsReadProtocolProcessor.cs:412]:
```suggestion
            _logger.LogWarning(NatsLogEvents.Protocol, "Reached invalid line");
```

File: src/NATS.Client.Core/NatsConnection.cs [hunk 2]
```diff
@@ -471,7 +471,7 @@ private async void ReconnectLoop()
             // If dispose this client, WaitForClosed throws OperationCanceledException so stop reconnect-loop correctly.
             await _socket!.WaitForClosed.ConfigureAwait(false);
 
-            _logger.LogTrace($"Detect connection {_name} closed, start to cleanup current connection and start to reconnect.");
+            _logger.LogTrace(NatsLogEvents.Connection, "Detect connection {Name} closed, start to cleanup current connection and start to reconnect", _name);
```
Comment (Jarema) [inline src/NATS.Client.Core/NatsConnection.cs:474]: Inconsistent tense with itself and other logs.

File: src/NATS.Client.Core/NatsConnection.cs [hunk 3]
```diff
@@ -519,7 +519,7 @@ private async void ReconnectLoop()
                         }
                     }
 
-                    _logger.LogInformation("Try to connect NATS {0}", url);
+                    _logger.LogInformation(NatsLogEvents.Connection, "Try to connect NATS {Url}", url);
```
Comment (Jarema) [inline src/NATS.Client.Core/NatsConnection.cs:522]:
```suggestion
                    _logger.LogInformation(NatsLogEvents.Connection, "Tried to connect NATS {Url}", url);
```
Inconsistency

File: src/NATS.Client.Core/NatsConnection.cs [hunk 4]
```diff
@@ -556,7 +556,7 @@ private async void ReconnectLoop()
             {
                 if (url != null)
                 {
-                    _logger.LogError(ex, "Fail to connect NATS {0}", url);
+                    _logger.LogError(NatsLogEvents.Connection, ex, "Fail to connect NATS {Url}", url);
```
Comment (Jarema) [inline src/NATS.Client.Core/NatsConnection.cs:559]:
```suggestion
                    _logger.LogError(NatsLogEvents.Connection, ex, "Failed to connect NATS {Url}", url);
```

File: src/NATS.Client.Core/NatsConnection.cs [hunk 5]
```diff
@@ -654,7 +654,7 @@ private async Task WaitWithJitterAsync()
         var waitTime = TimeSpan.FromMilliseconds(jitter) + backoff;
         if (waitTime != TimeSpan.Zero)
         {
-            _logger.LogTrace("Wait {0}ms to reconnect.", waitTime.TotalMilliseconds);
+            _logger.LogTrace(NatsLogEvents.Connection, "Wait {WaitMs}ms to reconnect", waitTime.TotalMilliseconds);
```
Comment (Jarema) [inline src/NATS.Client.Core/NatsConnection.cs:657]:
```suggestion
            _logger.LogTrace(NatsLogEvents.Connection, "Waiting {WaitMs}ms to reconnect", waitTime.TotalMilliseconds);
```

File: src/NATS.Client.Core/NatsConnection.cs [hunk 6]
```diff
@@ -672,7 +672,7 @@ private async void StartPingTimer(CancellationToken cancellationToken)
             {
                 if (Interlocked.Increment(ref _pongCount) > Opts.MaxPingOut)
                 {
-                    _logger.LogInformation("Detect MaxPingOut, try to connection abort.");
+                    _logger.LogInformation(NatsLogEvents.Connection, "Detect MaxPingOut, try to connection abort");
```
Comment (Jarema) [inline src/NATS.Client.Core/NatsConnection.cs:675]: Inconsistent with others. Not sure what the intention is.

**Review by Jarema — CHANGES_REQUESTED** [review]
Just some nits

File: src/NATS.Client.Core/NatsConnection.cs (hunk 2 above)
Comment (mtmk) [reply src/NATS.Client.Core/NatsConnection.cs]: Fixed as `Connection {Name} is closed. Will cleanup and reconnect`.
happy to hear other suggestions.

**Review by Jarema — APPROVED** [review]
LGTM

**Pull request merged**